File: fitness.jl
## Fitness Function
function fitness(pop,k,qobj,N_ind)
let
	fity = zeros(N_ind)
	file = "vocal_tract_tmp";
	for i = 1:N_ind
		#println("N_ind = ", i)
		Ac = pop[i,:]';
		qdom = vocal_tract(Ac,file,k);
		# fity[i] = vecnorm(qdom-qobj); # julia v0.6 
		#fity[i] = 100. /(relative_error*norm(qdom-qobj)); # julia v0.7>
		fity[i] = 80. /norm(qdom-qobj); # julia v0.7>
	end
	return fity
end
end

File: fitness.m
%% Fitness Function
% pop: population to be coded
% INPUTS: entries
% OUTPUTS: descedent of the generation

function [OUTPUT,fit] = fitness(pop,fit_function)
    
    % Pre-allocating memory
    Nind=length(pop(:,1));
    OUTPUT=zeros(Nind,1);
    fit=zeros(Nind,1);
    
    for i=1:Nind
        INPUT = pop(i,:);
        [~,OUTPUT(i)] = fit_function(INPUT);
        fit(i) = 1/(OUTPUT(i)+10);
    end
end